coffea/nanoevents/factory.py

```diff
@@ -87,71 +92,153 @@ def from_file(
+
+        return cls._from_mapping(mapping, partition_meta, base_form,
+                                 runtime_cache, persistent_cache,
+                                 schemaclass, metadata)
+
+    @classmethod
+    def from_parquet(
+        cls,
+        file,
+        treepath="/Events",
+        entry_start=None,
+        entry_stop=None,
+        runtime_cache=None,
+        persistent_cache=None,
+        schemaclass=NanoAODSchema,
+        metadata=None,
+        parquet_options={},
+        access_log=None,
+    ):
+        """Quickly build NanoEvents from a parquet file
+
+        Parameters
+        ----------
+            file : str, pathlib.Path, pyarrow.NativeFile, or python file-like
+                The filename or already opened file using e.g. ``uproot.open()``
+            treepath : str, optional
+                Name of the tree to read in the file
+            entry_start : int, optional
+                Start at this entry offset in the tree (default 0)
+            entry_stop : int, optional
+                Stop at this entry offset in the tree (default end of tree)
+            runtime_cache : dict, optional
+                A dict-like interface to a cache object. This cache is expected to last the
+                duration of the program only, and will be used to hold references to materialized
+                awkward1 arrays, etc.
+            persistent_cache : dict, optional
+                A dict-like interface to a cache object. Only bare numpy arrays will be placed in this cache,
+                using globally-unique keys.
+            schemaclass : BaseSchema
+                A schema class deriving from `BaseSchema` and implementing the desired view of the file
+            metadata : dict, optional
+                Arbitrary metadata to add to the `base.NanoEvents` object
+            parquet_options : dict, optional
+                Any options to pass to ``pyarrow.parquet.ParquetFile``
+            access_log : list, optional
+                Pass a list instance to record which branches were lazily accessed by this instance
+        """
+        ftypes = (str, pathlib.Path, pyarrow.NativeFile,
+                  io.TextIOBase, io.BufferedIOBase, io.RawIOBase, io.IOBase)
+        if isinstance(file, ftypes):
+            table_file = pyarrow.parquet.ParquetFile(file, **parquet_options)
+        elif isinstance(file, pyarrow.parquet.ParquetFile):
+            table_file = file
+        else:
+            raise TypeError("Invalid file type (%s)" % (str(type(file))))
+
+        if entry_start is None or entry_start < 0:
+            entry_start = 0
+        if entry_stop is None or entry_stop > table_file.metadata.num_rows:
+            entry_stop = table_file.metadata.num_rows
+
+        pqmeta = table_file.schema_arrow.metadata
+        pquuid = None if pqmeta is None else pqmeta.get(b'uuid', None)
+        pqurl = str(None if pqmeta is None else pqmeta.get(b'url', None))
+
+        partition_meta = (
```

Comment:
The idea is that the partition key, along with a uuidpfn map, can be used all by itself to locate and load the data. Obviously in a `from_file` situation that is trivial, but the idea is that a database holds only the partition keys.